pac4j-kerberos/src/test/java/org/pac4j/kerberos/client/direct/KerberosClientTests.java

```diff
@@ -0,0 +1,91 @@
+package org.pac4j.kerberos.client.direct;
+
+import static org.junit.Assert.assertEquals;
+import static org.mockito.Mockito.*;
+
+import java.io.UnsupportedEncodingException;
+import java.nio.charset.StandardCharsets;
+import java.util.Base64;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.pac4j.core.context.HttpConstants;
+import org.pac4j.core.context.J2EContext;
+import org.pac4j.core.context.MockWebContext;
+import org.pac4j.core.exception.DeferredHttpAction;
+import org.pac4j.core.exception.HttpAction;
+import org.pac4j.core.profile.CommonProfile;
+import org.pac4j.core.util.TestsConstants;
+import org.pac4j.core.util.TestsHelper;
+import org.pac4j.kerberos.credentials.KerberosCredentials;
+import org.pac4j.kerberos.credentials.authenticator.KerberosAuthenticator;
+import org.pac4j.kerberos.credentials.authenticator.KerberosTicketValidation;
+import org.pac4j.kerberos.credentials.authenticator.KerberosTicketValidator;
+
+
+
+public class KerberosClientTests implements TestsConstants {
+	
+    private KerberosAuthenticator kerberosAuthenticator;
+    private KerberosTicketValidator krbValidator;
+    
+    private final static byte[] KERBEROS_TICKET =  Base64.getEncoder().encode("Test Kerberos".getBytes(StandardCharsets.UTF_8));
+
+    @Before
+    public void before() {
+        // mocking
+        this.kerberosAuthenticator = mock(KerberosAuthenticator.class);
+        this.krbValidator = mock(KerberosTicketValidator.class);
+    }
+    
+    @Test
+    public void testMissingKerberosAuthenticator() {
+        final KerberosClient kerberosClient = new KerberosClient(null);
+        TestsHelper.initShouldFail(kerberosClient, "authenticator cannot be null");
+    }
+
+    @Test
+    public void testMissingProfileCreator() {
+        final KerberosClient kerberosClient = new KerberosClient(kerberosAuthenticator);
+        kerberosClient.setProfileCreator(null);
+        TestsHelper.initShouldFail(kerberosClient, "profileCreator cannot be null");
+    }
+
+    @Test
+    public void testBadAuthenticatorType() {
+        final KerberosClient kerberosClient = new KerberosClient(new DummyAuthenticator());

```

Comment:
As you only use the `DummyAuthenticator` once, I'd prefer to have a lambda here.